### PR TITLE
koordlet: add gc logs for metric cache

### PR DESCRIPTION
**Pull request by zwzhang0107**
### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/koordlet/metriccache/storage_test.go [hunk 1]
```diff
@@ -138,6 +138,14 @@ func Test_storage_PodResourceMetric_CRUD(t *testing.T) {
 				}
 			}
 
+			gotNum, err := s.CountPodResourceMetric()
+			if err != nil {
+				t.Errorf("CountPodResourceMetric got error %v", err)
```
Comment (saintube) [inline pkg/koordlet/metriccache/storage_test.go:143]: UT failed

**Review by saintube — CHANGES_REQUESTED** [review]

File: pkg/koordlet/metriccache/metric_cache.go [hunk 2]
```diff
@@ -642,7 +642,16 @@ func (m *metricCache) recycleDB() {
...
 	}
 	// raw records do not need to cleanup
-	klog.Infof("expired metric data before %v has been recycled", expiredTime)
+	nodeResCount, _ := m.db.CountNodeResourceMetric()
+	podResCount, _ := m.db.CountPodResourceMetric()
+	containerResCount, _ := m.db.CountContainerResourceMetric()
+	beCPUResCount, _ := m.db.CountBECPUResourceMetric()
+	podThrottledResCount, _ := m.db.CountPodThrottledMetric()
+	containerThrottledResCount, _ := m.db.CountContainerThrottledMetric()
+	klog.V(4).Infof("expired metric data before %v has been recycled, remaining in db size: "+
```
Comment (saintube) [inline pkg/koordlet/metriccache/metric_cache.go:651]: /lgtm

**Review by saintube — APPROVED** [review]

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Pull request merged**